File: src/main/kotlin/customfit/ai/kotlinclient/network/connection/ConnectionStatusListener.kt
package customfit.ai.kotlinclient.network.connection

/** Interface for objects that will be notified of connection status changes. */
interface ConnectionStatusListener {
    /**
     * Called when the connection status changes.
     *
     * @param newStatus The new connection status
     * @param info Detailed connection information
     */
    fun onConnectionStatusChanged(newStatus: ConnectionStatus, info: ConnectionInformation)
}

File: customfit-swift-client-sdk/Sources/CustomFitSwiftSDK/Network/Connection/ConnectionStatusListener.swift
import Foundation

// This file imports the ConnectionStatusListener protocol from ListenerManager
// to avoid duplication and ambiguity.
@_exported import struct Foundation.URL
@_exported import protocol CustomFitSwiftSDK.ConnectionStatusListener

/// Connection status listener protocol matching Kotlin implementation
public protocol ConnectionStatusListener: AnyObject {
    /**
     * Called when the connection status changes.
     *
     * @param newStatus The new connection status
     * @param info Detailed connection information
     */
    func onConnectionStatusChanged(newStatus: ConnectionStatus, info: ConnectionInformation)
} 